lib/src/main/java/io/ably/lib/rest/Auth.java

```diff
@@ -616,8 +617,12 @@ public TokenDetails requestToken(TokenParams params, AuthOptions tokenOptions) t
                     signedTokenRequest = (TokenRequest)authCallbackResponse;
                 else
                     throw AblyException.fromErrorInfo(new ErrorInfo("Invalid authCallback response", 400, 40000));
-            } catch(AblyException e) {
-                throw AblyException.fromErrorInfo(e, new ErrorInfo("authCallback failed with an exception", 401, 80019));
+            } catch (Exception e) {
+                boolean isTokenExceptionNonRetriable = e instanceof NonRetriableTokenException;
+                boolean isAblyExceptionNonRetriable = e instanceof AblyException && ((AblyException) e).errorInfo.statusCode == 403;
+                boolean shouldNotRetryAuthOperation = isTokenExceptionNonRetriable || isAblyExceptionNonRetriable;
+                int statusCode = shouldNotRetryAuthOperation ? 403 : 401; // RSA4c & RSA4d
+                throw AblyException.fromErrorInfo(e, new ErrorInfo("authCallback failed with an exception", statusCode, 80019));
```

Comment:
We should extract Ably error codes to AblyConstants as well and link them to a specific API doc page.

Reply:
The same thing as HTTP constants, we will do this in the near future. https://github.com/ably/ably-java/issues/838